cpp/cugraph_module/algorithms/betweenness_centrality.cu

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2016-2022 Memgraph Ltd. [https://memgraph.com]
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "mg_cugraph_utility.hpp"
+
+namespace {
+// TODO: Check Betweenness instances
+using vertex_t = int32_t;
+using edge_t = int32_t;
+using weight_t = double;
+using result_t = double;
+
+constexpr char const *kProcedureBetweenness = "get";
+
+constexpr char const *kArgumentNormalized = "normalized";
+constexpr char const *kArgumentDirected = "directed";
+constexpr char const *kArgumentWeightProperty = "weight";
+
+constexpr char const *kResultFieldNode = "node";
+constexpr char const *kResultFieldBetweennessCentrality = "betweenness_centrality";
+
+const double kDefaultWeight = 1.0;
+
+void InsertBetweennessRecord(mgp_graph *graph, mgp_result *result, mgp_memory *memory, const std::uint64_t node_id,
+                             double rank) {
+  auto *record = mgp::result_new_record(result);
+  mg_utility::InsertNodeValueResult(graph, record, kResultFieldNode, node_id, memory);
+  mg_utility::InsertDoubleValueResult(record, kResultFieldBetweennessCentrality, rank, memory);
+}
+
+void BetweennessProc(mgp_list *args, mgp_graph *graph, mgp_result *result, mgp_memory *memory) {
+  try {
+    auto normalized = mgp::value_get_bool(mgp::list_at(args, 0));
+    auto directed = mgp::value_get_bool(mgp::list_at(args, 1));
+    auto weight_property = mgp::value_get_string(mgp::list_at(args, 2));
+
+    raft::handle_t handle{};
+    auto stream = handle.get_stream();
+
+    auto mg_graph = mg_utility::GetWeightedGraphView(
+        graph, result, memory, directed ? mg_graph::GraphType::kDirectedGraph : mg_graph::GraphType::kUndirectedGraph,
+        weight_property, kDefaultWeight);
+    if (mg_graph->Empty()) return;
+
+    auto n_vertices = mg_graph.get()->Nodes().size();
+    // IMPORTANT: Betweenness centrality cuGraph algorithm works only on legacy code
+    auto cu_graph_ptr =
+        mg_cugraph::CreateCugraphLegacyFromMemgraph<vertex_t, edge_t, weight_t>(*mg_graph.get(), handle);
+    auto cu_graph_view = cu_graph_ptr->view();
+    cu_graph_view.prop.directed = directed;
```

Comment:
I would just put a comment here that this is a hack, and we users should be aware.

Reply:
This is how it was done in Legacy module, so till the change to new API, this is actually not the hack but the way of using the API...

Reply:
Okay

Reply:
Should we then put a comment about that?

Reply:
There is a note called `IMPORTANT` in each occurrence of Legacy Graph instantiation.